lib/string_map.h

```diff
@@ -23,7 +23,20 @@ limitations under the License.
 #include "absl/container/flat_hash_map.h"
 #include "cstring.h"
 
-// Map is ordered by order of element insertion.
+/// @file
```

Comment:
Maybe associate the comment with the class instead? I'm not sure what best practices for doxygen we have, but it seems to me it would make the comment more visible.